### PR TITLE
👌 IMPROVE: put composer attachment buttons into three dot menu

### PR DESCRIPTION
**Pull request by violoncelloCH**
as discussed with @jancborchardt

### CODE REVIEW

**Comment by ChristophWurst** [comment]
> as discussed with @jancborchardt

For https://github.com/nextcloud/mail/pull/2222#issuecomment-552147470?

File: src/components/Composer.vue [hunk 1]
```diff
@@ -348,6 +357,12 @@ export default {
 		onInputChanged() {
 			this.saveDraftDebounced(this.getMessageData())
 		},
+		onAddLocalAttachment() {
+			this.$refs.composerAttachments.onAddLocalAttachment()
```
Comment (ChristophWurst) [inline src/components/Composer.vue]: In general not a big fan of this coding style. This should be doable with events. Check https://stackoverflow.com/a/42634865/2239067 and https://stackoverflow.com/questions/43256978/event-from-parent-to-child-component/43258428#43258428. The former is less code, but requires a parent component (problematic for isolated testing), the latter requires a bit more code but is quite elegant and works with testing :)

File: src/components/Composer.vue [hunk 2]
```diff
@@ -123,17 +123,26 @@
...
-					}}</ActionCheckbox>
+					<ActionButton icon="icon-upload" @click="onAddLocalAttachment">{{
+						t('mail', 'Upload attachment')
+					}}</ActionButton>
+					<ActionButton icon="icon-folder" @click="onAddCloudAttachment">{{
+						t('mail', 'Add attachment from Files')
+					}}</ActionButton>
+					<ActionCheckbox
+						:checked="!editorPlainText"
+						:text="t('mail', 'Enable text formatting')"
```
Comment (ChristophWurst) [inline src/components/Composer.vue]: Please revert to the original message.

Moving the checkbox is okay, but sneaking in a new label with an unrelated change is a bit meh. Did @jancborchardt talk you into this? :D

**Comment by violoncelloCH** [comment]
> > as discussed with @jancborchardt
> 
> For [#2222 (comment)](https://github.com/nextcloud/mail/pull/2222#issuecomment-552147470)?

no, actually this one is about the composer but #2222 is about the reading/displaying view...
this one is rather for #2220 to further improve the mobile composer view (and in general keep the interface clean)

File: src/components/Composer.vue (hunk 1 above)
Comment (violoncelloCH) [reply src/components/Composer.vue]: oh, I didn't even know that this way of doing it existed...
maybe it would even make sense to handle this in a vuex store? :thinking: 

anyway I'm happy to implement it the way you want :)

File: src/components/Composer.vue (hunk 2 above)
Comment (violoncelloCH) [reply src/components/Composer.vue]: hmm, @jancborchardt and me thought it would make sense to change this, because a checkbox saying "plain text" which reveals formatting possibilities when unchecked is only understandable for techie users. For accessibility reasons it would make sense to describe features in clear and precise language...
but of course if you want me to revert this I can do so or I could also create a separate PR to discuss this there if you prefer...?

Comment (jancborchardt) [reply src/components/Composer.vue]: It was @violoncelloch's suggestion and I agree with it, also saying it could be added here since we talked about it in the same context. Yes it should be in a separate PR, my mistake. :)

Comment (ChristophWurst) [reply src/components/Composer.vue]: Yes, @violoncelloCH, please do this in a separate PR. I'm fine with the change, just not when the PR is about *put composer attachment buttons into three dot menu* :wink:

Comment (violoncelloCH) [reply src/components/Composer.vue]: let me do that :)

File: src/components/Composer.vue (hunk 1 above)
Comment (violoncelloCH) [reply src/components/Composer.vue]: so I implemented the second variant with the local event bus :)
hope this is fine now...

Comment (ChristophWurst) [reply src/components/Composer.vue]: I'd only use Vuex for global state. These attachments are local state, only really relevant to these components.

File: src/components/Composer.vue [hunk 3]
```diff
@@ -348,6 +358,16 @@ export default {
 		onInputChanged() {
 			this.saveDraftDebounced(this.getMessageData())
 		},
+		onAddLocalAttachment() {
+			this.bus.$emit('event', {
```
Comment (ChristophWurst) [inline src/components/Composer.vue]:
```suggestion
			this.bus.$emit('onAddLocalAttachment', {
```

File: src/components/ComposerAttachments.vue [hunk 4]
```diff
@@ -136,6 +133,16 @@ export default {
 			this.$emit('input', this.value.filter(a => a !== attachment))
 		},
 	},
+	created() {
+		this.bus.$on('event', (args) => {
```
Comment (ChristophWurst) [inline src/components/ComposerAttachments.vue]:
```suggestion
		this.bus.$on('onAddLocalAttachment', (args) => {
```

:wink:

File: src/components/ComposerAttachments.vue [hunk 5]
```diff
@@ -80,6 +77,14 @@ export default {
 			return ((uploaded / total) * 100).toFixed(1)
 		},
 	},
+	created() {
+		this.bus.$on('onAddLocalAttachment', args => {
+			this.onAddLocalAttachment()
+		})
+		this.bus.$on('onAddCloudAttachment', args => {
+			this.onAddCloudAttachment()
```
Comment (violoncelloCH) [inline src/components/ComposerAttachments.vue]: thanks @ChristophWurst ! Implemented your changes...
only thing I wasn't sure is if I could omit those `args` somehow which aren't used

Comment (ChristophWurst) [reply src/components/ComposerAttachments.vue]: Yes! ``this.bus.$on('onAddCloudAttachment', this.onAddCloudAttachment)``

Comment (violoncelloCH) [reply src/components/ComposerAttachments.vue]: ahh, makes sense... thank you!
